lte/cloud/go/services/subscriberdb/storage/digest.go

```diff
@@ -0,0 +1,222 @@
+		}
+
+		rows, err := l.builder.
+			Select(digestLookupNidCol, digestLookupDigestCol, digestLookupLastUpdatedTimeCol).
+			From(digestLookupTableName).
+			Where(filters).
+			RunWith(tx).
+			Query()
+		if err != nil {
+			return nil, errors.Wrapf(err, "gets digest for networks %+v", networks)
+		}
+		defer sqorc.CloseRowsLogOnError(rows, "GetDigest")
+
+		digestInfoSet := DigestInfoSet{}
+		for rows.Next() {
+			network, digest, lastUpdatedTime := "", "", int64(0)
+			err = rows.Scan(&network, &digest, &lastUpdatedTime)
+			if err != nil {
+				return nil, errors.Wrap(err, "select digests for networks, SQL row scan error")
+			}
+			digestInfo := digestInfo{
+				network:         network,
+				digest:          digest,
+				lastUpdatedTime: lastUpdatedTime,
+			}
+			digestInfoSet = append(digestInfoSet, digestInfo)
+		}
+		err = rows.Err()
+		if err != nil {
+			return nil, errors.Wrap(err, "select digests for network, SQL rows error")
+		}
+		return digestInfoSet, nil
+	}
+
+	txRet, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	if err != nil {
+		return nil, err
+	}
+	ret := txRet.(DigestInfoSet)
+	return ret, nil
+}
+
+func (l *digestLookup) SetDigest(network string, digest string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		sc := squirrel.NewStmtCache(tx)
+		defer sqorc.ClearStatementCacheLogOnError(sc, "SetDigest")
+
+		now := clock.Now().Unix()
+		_, err := l.builder.
+			Insert(digestLookupTableName).
+			Columns(digestLookupNidCol, digestLookupDigestCol, digestLookupLastUpdatedTimeCol).
+			Values(network, digest, now).
+			OnConflict(
+				[]sqorc.UpsertValue{
+					{Column: digestLookupDigestCol, Value: digest},
+					{Column: digestLookupLastUpdatedTimeCol, Value: now},
+				},
+				digestLookupNidCol,
+			).
+			RunWith(sc).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "insert digest for network %+v", network)
+		}
+		return nil, nil
+	}
+
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func (l *digestLookup) DeleteDigests(networks []string) error {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		_, err := l.builder.
+			Delete(digestLookupTableName).
+			Where(squirrel.Eq{digestLookupNidCol: networks}).
+			RunWith(tx).
+			Exec()
+		if err != nil {
+			return nil, errors.Wrapf(err, "delete digests")
+		}
+		return nil, nil
+	}
+
+	_, err := sqorc.ExecInTx(l.db, nil, nil, txFn)
+	return err
+}
+
+func GetDigest(l DigestLookup, network string) (string, int64, error) {
+	digestInfoSet, err := l.GetDigests([]string{network}, 0)
+	if err != nil {
+		return "", 0, err
+	}
+	// There should be at most 1 digest for each network
+	// if digest not found, return default value
+	if len(digestInfoSet) == 0 {
+		return "", 0, nil
+	}
+	digestInfo := digestInfoSet[0]
+	return digestInfo.digest, digestInfo.lastUpdatedTime, nil
+}
+
+func GetOutdatedNetworks(l DigestLookup, updateDeadlineTime int64) ([]string, error) {
```

Comment:
From previous comment: provide a default, public const in this file that callers can pass as the updatedDeadlineTime

Reply:
Edit to the "provide a default, public const" comment: based on the way you're using this, I see that it's not actually necessary. So yeah can ignore that comment, no need for a const